src/middlewares/requiredFieldsMiddleware.ts

```diff
@@ -38,7 +38,14 @@ const requiredFields = ({ getState, dispatch }: MiddlewareAPI<Dispatch<AnyAction
             state.view.widgets
             .filter(item => item.bcName === widget.bcName)
             .forEach(item => {
-                item.fields.forEach((widgetField: WidgetField) => {
+                let itemFieldsCalc: object[] = []
+                if (item.type === WidgetTypes.List) {
+                    itemFieldsCalc = item.fields
+                }
+                if (item.type === WidgetTypes.Form) {
+                    item.fields.forEach((block: WidgetBlock) => {block.fields.forEach((field: []) => itemFieldsCalc.push(field))})
```

Comment:
At the moment it's not mandatory for Form widget fields to be a block:
https://github.com/tesler-platform/tesler-ui/blob/056ae71c4ae3313e5b87b96d0c8dd2d1ddd541e6/src/interfaces/widget.ts#L203-L208

We should check if it's really a `WidgetBlock` with a Type Guard:
https://github.com/tesler-platform/tesler-ui/blob/056ae71c4ae3313e5b87b96d0c8dd2d1ddd541e6/src/interfaces/widget.ts#L240
There is a [usage example](https://github.com/tesler-platform/tesler-ui/blob/056ae71c4ae3313e5b87b96d0c8dd2d1ddd541e6/src/hooks/useFlatFormFields.ts#L10) for this Type Guard